src/modules/dpos/validator/components/UnlockBalanceForm/unlockBalance.test.js

```diff
@@ -96,32 +96,83 @@ describe('Unlock LSK modal', () => {
     },
   };
 
+  const rawTx = {
+    asset: {
+      unlockObjects: [
+        {
+          amount: '3000000000',
+          delegateAddress: 'lskdwsyfmcko6mcd357446yatromr9vzgu7eb8y11',
+          unvoteHeight: 100,
+        },
+        {
+          amount: '1000000000',
+          delegateAddress: 'lskdwsyfmcko6mcd357446yatromr9vzgu7eb8y13',
+          unvoteHeight: 3000,
+        },
+      ],
+    },
+    fee: 10000000,
+    moduleAssetId: '5:2',
+    nonce: '178',
+    sender: {
+      publicKey: '0fe9a3f1a21b5530f27f87a414b549e79a940bf24fdf2b2f05e7f22aeeecc86a',
+    },
+  };
+
   beforeEach(() => {
-    wrapper = mountWithProps(LockedBalance, props, store);
+    wrapper = mountWithProps(UnlockBalanceForm, props, store);
     hwManager.signTransactionByHW.mockResolvedValue({});
   });
 
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
   it('renders the LockedBalance component properly', () => {
     expect(wrapper).toContainMatchingElement('.lock-balance-amount-container');
     expect(wrapper).toContainMatchingElement('.transaction-priority');
-    expect(wrapper).toContainMatchingElement('.unlock-btn');
+    expect(wrapper).toContainMatchingElement('.confirm-btn');
   });
 
   it('fires balanceUnlocked action with selected fee', async () => {
     const tx = { id: 1 };
-    create.mockImplementation(() =>
+    createGenericTx.mockImplementation(() =>
       new Promise((resolve) => {
         resolve(tx);
       }));
 
     // Act
-    wrapper.find('.unlock-btn').at(0).simulate('click');
+    wrapper.find('.confirm-btn').at(0).simulate('click');
     act(() => { wrapper.update(); });
     await flushPromises();
-    expect(props.nextStep).toBeCalledWith({
-      rawTransaction: {
-        selectedFee: '0.1',
+    expect(props.nextStep).toBeCalledWith({ rawTx });
+  });
+
+  it('calls nextStep when clicked on confirm', async () => {
+    wrapper.find('.confirm-btn button').simulate('click');
+    expect(props.nextStep).toBeCalledWith(
+      expect.objectContaining({ rawTx }),
+    );
+  });
+
+  it.skip('does not call nextStep when unlockableBalance is zero', async () => {
```

Comment:
was this by mistake? can we un-skip it?

Reply:
No, there's an issue with the mocks. I could change it, but it would cause many other tests break. We can fix it in another time